### DISCUSSION TITLE
Multiple Themes

### DISCUSSION THREAD

**Discussion by Lplatedeveloper** [mkdocs/mkdocs#3645] [Q&A] [answered]
Is it possible to have multiple custom themes for the 1 set of documentation files?  and a different nav structure for each theme?
We have various partners that want there branding to appear.
I don't really want to create a seperate documentation files for each.

**Comment by lovelydinosaur** [comment, marked as answer]
You could have this kind of layout...

```shell
site1.yml
site2.yml
site3.yml
docs/...
```

`mkdocs -f site1.yml`

Each config file could also point to a [different custom theme directory](https://www.mkdocs.org/user-guide/customizing-your-theme/#using-the-theme-custom_dir).

**Reply by Andre601** [reply]
Your aproach has the issue of targeting the site as a whole, which may not be what OP wants.
Plus it's a lot of extra work.

I personally would've chosen to override the main file that adds the css to the pages and added something like this:
```jinja2
{% if page.meta and page.meta.custom_css %}
    {# add the css #}
{% else %}
    {# Add default css #}
{% endif %}
```

**Reply by Lplatedeveloper** [reply]
Thank you tomchristie and Andre601.
Unfortunately, I'm no developer.
@Andre601, what do you mean by the main file?  where abouts in the main file would I add that.  Could you provide an example?

**Reply by Andre601** [reply]
I talk about theme extension. Every theme has a main file they use as their base.
As an example, the Material for MkDocs theme has a base.html file containing all the content of a basic site and a main.html that extends it.
If you now want to edit the page itself across your site, you would override the theme using the `theme.custom_dir` option in the mkdocs.yml file, add a main.html and override the parts you want.

Tho, this is quite complicated and what I gave now was a severe simplification. I suggest learning about extending themes and the jinja2 templating language yourself, as I cannot provide a decent tutorial on this.

**Reply by carlocastoldi** [reply]
FYI i made a simple plugin for this: https://github.com/carlocastoldi/mkdocs-multi-theme-plugin

however it needs https://github.com/mkdocs/mkdocs/pull/3786 to be merged (or my forked version of mkdocs)

**Reply by kamilkrzyskow** [reply]
`page.meta.template` can be set in any event which has access to the `page` and that comes before `on_env`, so `on_page_markdown` is a good candidate, especially as it's now the go to event for meta mangling after `on_page_read_source` has been deprecated in 1.6.0

The `template.html` files can also be edited on the fly, which I've done in my hook here:
- https://github.com/squidfunk/mkdocs-material/discussions/5060

But overall more events, or more params in each event would be welcome to be able to do things in a less hacky way ;]
